### PR TITLE
Add Test for Multipage RGB TIFF

### PR DESCRIPTION
**Pull request by blink1073**
Also uncovered a couple of subtle bugs in regards to indexing multi-page images.
Previously, we were returning the *all* images when `img_num` was 0.

### CODE REVIEW

File: skimage/io/tests/test_multi_image.py [hunk 1]
```diff
@@ -14,7 +15,8 @@ class TestMultiImage():
     def setUp(self):
         # This multipage TIF file was created with imagemagick:
         # convert im1.tif im2.tif -adjoin multipage.tif
-        paths = [os.path.join(data_dir, 'multipage.tif'),
+        use_plugin('pil')
+        paths = [os.path.join(data_dir, 'multipage_rgb.tif'),
```
Comment (jni) [inline skimage/io/tests/test_multi_image.py]: Do we actually want to test this just on PIL, or test both PIL and tifffile?

**Comment by jni** [comment]
@blink1073 I like this, but I'm not sure about the `use_plugin` directive, rather that trying that call twice, once with `plugin='PIL'` and once with `plugin='tifffile'`. (I may be messing up the kwarg syntax here, but you get my drift.)

**Comment by blink1073** [comment]
@jni, I had added that while debugging and forgotten about it.  I'd just as soon leave it out.

**Comment by jni** [comment]
@blink1073 why? It looks like it catches specific bugs? Since Pillow is a dependency, we might as well test it here also?

**Comment by blink1073** [comment]
Actually, if we aren't specific, it will use whichever plugin was previously used in another file, which could cause a bug.  We can't use `tifffile` because there is a `.gif` in there.  I'm just going to put `pil` back.

File: skimage/io/tests/test_pil.py [hunk 2]
```diff
@@ -50,6 +50,24 @@ def test_imread_flatten():
...
+    f.close()
+    imsave(fname, x)
+    img = imread(fname)
+    os.remove(fname)
+    assert img.shape == (16, 8, 3), img.shape
+
+
+def test_imread_multipage_rgb_tif():
+    img = imread(os.path.join(data_dir, 'multipage_rgb.tif'))
+    assert img.shape == (2, 10, 10, 3), img.shape
```
Comment (jni) [inline skimage/io/tests/test_pil.py]: why is there a `img.shape` at the end?

Also, can you add this test to `test_tifffile.py` as well? Seems like a good test to have. Currently the tifffile plugin doesn't have a multipage test, that I can tell.

Comment (blink1073) [inline skimage/io/tests/test_pil.py]: The second argument is what gets printed if the assert fails, it is a debug statement.  I agree about `test_tifffile`.

Comment (blink1073) [inline skimage/io/tests/test_pil.py]: Done.

**Comment by jni** [comment]
Thanks @blink1073! =)

**Pull request merged**